File: CS/MySolution.Module/Security/CustomAuthenticationStandardProvider.cs
using DevExpress.ExpressApp;
using DevExpress.ExpressApp.Security;
using MySolution.Module.BusinessObjects;
using System;

namespace MySolution.Module.Security {

    // ...
    public class CustomAuthenticationStandardProvider : AuthenticationStandardProvider {
        public CustomAuthenticationStandardProvider(Type userType) : base(userType) {
        }
        public override object Authenticate(IObjectSpace objectSpace) {
            ApplicationUser user = base.Authenticate(objectSpace) as ApplicationUser;
            if (user != null && !user.EnableStandardAuthentication) {
                throw new InvalidOperationException("Password authentication is not allowed for this user.");
            }
            return user;
        }
    }
}

File: VB/MySolution.Module/Security/CustomAuthenticationStandardProvider.vb
Imports DevExpress.ExpressApp
Imports DevExpress.ExpressApp.Security
Imports MySolution.Module.BusinessObjects
Imports System

Namespace MySolution.Module.Security

	' ...
	Public Class CustomAuthenticationStandardProvider
		Inherits AuthenticationStandardProvider

		Public Sub New(ByVal userType As Type)
			MyBase.New(userType)
		End Sub
		Public Overrides Function Authenticate(ByVal objectSpace As IObjectSpace) As Object
			Dim user As ApplicationUser = TryCast(MyBase.Authenticate(objectSpace), ApplicationUser)
			If user IsNot Nothing AndAlso Not user.EnableStandardAuthentication Then
				Throw New InvalidOperationException("Password authentication is not allowed for this user.")
			End If
			Return user
		End Function
	End Class
End Namespace
